command/agent/config.go

```diff
@@ -1437,6 +1437,20 @@ func FixupCheckType(raw interface{}) error {
 		case "service_id":
 			rawMap["serviceid"] = v
 			delete(rawMap, k)
+		case "header":
+			vm, ok := v.(map[string]interface{})
```

Comment:
I'll add that but I want to have a closer look at this method in general since it looks like mangling `map[string]interface{}` is the wrong approach to this. But that could be fixed with the decoupling of user and runtime config where user config represents exactly what the user is providing and then we just parse the JSON and interpret it afterwards instead of mangling it.